### DISCUSSION TITLE
Contribute to Vercel Guides Page

### DISCUSSION THREAD

**Discussion by dhairyathedev** [vercel/community#915] [Show & Tell]
I would like to fix the broken code snippet on this page - https://vercel.com/guides/add-auth-to-nextjs-with-magic

Are the Vercel guides pages opensource?

**Comment by amyegan** [comment]
Unfortunately, the guides are not open at the moment. But I would be happy to submit changes for you if there is something incorrect in the document.

**Comment by dhairyathedev** [comment]
@amyegan 
**In the last code snippet of the page:** 

<img width="886" alt="Screenshot 2022-09-30 at 9 46 48 PM" src="https://user-images.githubusercontent.com/65452005/193313332-5c67a2f7-43c2-4af0-bc49-5df0cc8dfa65.png">

`<Link />` component is used without importing. So the snippet will be something like this:
```jsx
// pages/index.js

import Head from 'next/head';
import Link from 'next/link'

export default function Home() {
  return (
    <>
      <Head>
        <script
          dangerouslySetInnerHTML={{
            __html: `
          if (document.cookie && document.cookie.includes('authed')) {
            window.location.href = "/dashboard"
          }
        `,
          }}
        />
      </Head>
      <Link href="/login">
        <a>Login</a>
      </Link>
    </>
  );
}
```

**Comment by amyegan** [comment]
Thank you for pointing out the problem with the original code example. The guide and code examples were updated to include the necessary imports and make use of new features 😄
